### DISCUSSION TITLE
How to put a file of some contentType into the `pageBundle` of another contentType on creation?

### DISCUSSION THREAD

**Discussion by pauko** [estruyf/vscode-front-matter#715] [Q&A] [answered]
Hello community!

In my metamodel of records I have the entities _Solution_, _Problem_, _Opportunity_, _Goal_, whereas _Solution_ is some kind of container, which "aggregates" all other entities. Let's visualize this for better understanding with a class diagram:

```mermaid
classDiagram
  class Solution
  class Problem
  class Opportunity
  class Goal

  Solution --o Problem
  Solution --o Opportunity
  Solution --o Goal
``` 

In fact, in the `.md` files I have a `contentRelationship` from the other entities towards _Solution_ by slugs - so the reference goes from child to parent!

I want to establish a structure in my content folder, so that each _Solution_ gets its own subdirectory in some `contents` directory, and all _Problems_, _Opportunities_, and _Goals_ that are associated with some _Solution_ are placed in that corresponding subdirectory - ideally on creation of the `.md` file when setting the reference to the _Solution_.

```
contents/
├─ Solution A/
│ ├── index.md
│ ├── problem-1-of-A.md
│ ├── problem-2-of-A.md
│ ├── goal-1-of-A.md
├─ Solution B/
│ ├── index.md
│ ├── problem-1-of-B.md
...
```
My first idea is to use `pageBundle: true` for the _Solution_ content type, which would give me the subdirectories (and an `index.md`).

However, how can I configure the other entities, so that when I set the `contentRelationship` to the _Solution_ in the frontmatter of some `.md`, this `.md` file is placed into the subdirectory of that _Solution_?

I read the [Placeholders](https://frontmatter.codes/docs/content-creation/placeholders) section of the documentation, but I don't find a good fit - and I'm a little bit hesitating to go for the [custom placeholders](https://frontmatter.codes/docs/content-creation/placeholders#custom-placeholders) with their scripting abilities.

Is there something less sophisticated for my use case?

**Comment by estruyf** [comment]
Hi @pauko, that is an exciting structure you have there, and if I had the same use case, I would approach it the same way.

The `solution` is the easiest: creating a new page as a page bundle. The other types are a bit trickier, as they need to know the parent folder they need to make. There is nothing yet in Front Matter CMS that allows you to accomplish this with the default features.

What could work is the `postScript` functionality on the content type, which allows you to run a script after the content is created. So you can move it to the correct folder. - https://frontmatter.codes/docs/content-creation/content-types#run-a-script-after-your-content-is-created

Feel free to give suggestions on what would help you to get your content created.

What I could think of is that we can introduce a `subContent` property on the content type where you can specify the content to look for and propose that during the content creation.

Example:

When you want to create a problem, FM proposes to you: `Under which <content type> would you like to create the content: ...`

**Reply by pauko** [reply]
Hi @estruyf 

Well, I think, this semantics of `subContent` could be bundled with the `contentRelationship`, which is set in such a situation anyway.
For example, new fields associated with `contentRelationship` (in addition to `contentTypeName` and `contentTypeValue`) specify, if a `.md` file should be placed in the same directory of the `.md` file (content type) it refers to, or some sub-directory relative to it.

Maybe it also increases clarity and overview, if you pack those new fields and the existing ones into a dedicated object, like:
```json
{
...
"type": "contentRelationship"
"contentArgs": {
 " name": "<what was contentTypeName before>"
  "value": "<what was contentTypeValue before>"
  "location": ["same as related", "sub-directory XYZ of related", "original page folder"]
  }
...
}
```
It should not necessarily depend on `pageBundle: true`.
That says, if we set `"location": "same as related"`:
  - if `pageBundle: true` the `.md` file is placed in the page bundle directory
  - if `pageBundle: false` the `.md` file goes into the page folder of the related content type, which might be different from the page folder of the actual content type.

In other words, `location` gets priority over the `path` in some `pageFolder` configuration.

**Reply by pauko** [reply]
And, of course, it would be nice if this semantics of `subContent` worked recursively, thus building a hierarchy of subdirectories on the filesystem.

**Reply by estruyf** [reply]
During the content creation, we are not yet checking the fields. There is only validation on the content-type and its properties.

What you are proposing is to add some more logic to the `contentRelationship` field, which I get, but can also be tricky when configuring multiple fields. Another issue I could see is when adding it to the field, would it mean we have to move the file each time the option is changed?

When we would add it to the root of the content-type, it would only be used once for placing the file in the correct location.

**Reply by estruyf** [reply]
https://github.com/estruyf/vscode-front-matter/issues/722

**Reply by estruyf** [reply, marked as answer]
I've done the first steps to allow you to select the folder where the sub-content must be created. Check issue #722 to see how you can configure it. You can give it a try in the latest beta version.
